File: fathersDay/Dsa/Day 19 (832. Flipping an Image)/Solution.java
class Solution {
    public int[][] flipAndInvertImage(int[][] image) {
        for (int[] row : image) {
            int left = 0;
            int right = row.length - 1;
            while (left <= right) {
                // Swap and invert
                int temp = row[left] == 0 ? 1 : 0;
                row[left] = row[right] == 0 ? 1 : 0;
                row[right] = temp;
                left++;
                right--;
            }
        }
        return image;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int[][] image = {{1, 1, 0}, {1, 0, 1}, {0, 0, 0}};
        int[][] result = solution.flipAndInvertImage(image);
        for (int[] row : result) {
            for (int pixel : row) {
                System.out.print(pixel + " ");
            }
            System.out.println();
        }
        // Expected Output:
        // 1 0 0 
        // 0 1 0 
        // 1 1 1 
    }
}

File: fathersDay/Dsa/Day 19 (832. Flipping an Image)/Solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class Solution {
public:
    std::vector<std::vector<int>> flipAndInvertImage(std::vector<std::vector<int>>& image) {
        for (auto& row : image) {
            std::reverse(row.begin(), row.end());
            for (int& pixel : row) {
                pixel = 1 - pixel;
            }
        }
        return image;
    }
};

int main() {
    Solution solution;
    std::vector<std::vector<int>> image = {{1, 1, 0}, {1, 0, 1}, {0, 0, 0}};
    std::vector<std::vector<int>> result = solution.flipAndInvertImage(image);
    for (const auto& row : result) {
        for (int pixel : row) {
            std::cout << pixel << " ";
        }
        std::cout << std::endl;
    }
    // Expected Output:
    // 1 0 0 
    // 0 1 0 
    // 1 1 1 
    return 0;
}
